### ISSUE TITLE
Support ZeRO Stage 1 & 3

### ISSUE THREAD

**Issue by RobotSail** [instructlab/training#26]
Today we hardcode options specific to ZeRO stage 2. We should update our implementation
to allow support for ZeRO stage 1 and 3 as well.

**Comment by fabianlim** [comment]
If we do this, then all the checkpointing flows need to be retested. Also not sure what is the impact on #25

**Comment by RobotSail** [comment]
Yes, we shouldn't do this until after the 15th. But probably something we should eventually support if we'd like NVMe offloading.
